libbeat/feature/feature.go

```diff
@@ -0,0 +1,120 @@
+package feature
+
+import (
+	"fmt"
+	"reflect"
+)
+
+//go:generate stringer -type=Stability
+
+// Registry is the global plugin registry, this variable is mean to be temporary to move all the
+// internal factory to receive a context that include the current beat registry.
+var Registry = newRegistry()
+
+// Featurable implements the description of a feature.
+type Featurable interface {
+	// Namespace returns the namespace of the Feature.
+	Namespace() string
+
+	// Name returns the name of the feature. The name must be unique for each namespace.
+	Name() string
+
+	// Factory returns the factory func.
+	Factory() interface{}
+
+	// Stability returns the stability of the feature.
+	Stability() Stability
+
+	// Equal returns true if the two object are equal.
+	Equal(other Featurable) bool
+
+	String() string
+}
```

Comment:
'X returns x' is kind of redundant. Plus godoc will not generate any docs for the methods. Instead the type will be shown in the docs as is in code. Rather then stating the obvious, describing how `Featurable` interacts with the Bundle and Registry would be more helpful. What's the purpose of Namespace. Why do I need to implement 'equal'?